src/Graphql/Resolver/CollectionResolverFactory.php

```diff
@@ -96,18 +97,52 @@ public function createCollectionResolver(string $resourceClass, string $rootClas
 
             $data = ['edges' => [], 'pageInfo' => ['endCursor' => null, 'hasNextPage' => false]];
             if ($collection instanceof PaginatorInterface && ($totalItems = $collection->getTotalItems()) > 0) {
-                $data['pageInfo']['endCursor'] = \base64_encode((string) ($totalItems - 1));
+                $data['pageInfo']['endCursor'] = base64_encode((string) ($totalItems - 1));
                 $data['pageInfo']['hasNextPage'] = $collection->getCurrentPage() !== $collection->getLastPage() && (float) $collection->count() === $collection->getItemsPerPage();
             }
 
             foreach ($collection as $index => $object) {
                 $data['edges'][$index] = [
-                    'node' => $this->normalizer->normalize($object, null, ['graphql' => true] + $context),
-                    'cursor' => \base64_encode((string) ($index + $offset)),
+                    'node' => $this->normalizer->normalize($object, ItemNormalizer::FORMAT, $normalizationContext),
+                    'cursor' => base64_encode((string) ($index + $offset)),
                 ];
             }
 
             return $data;
         };
     }
+
+    private function fieldsToAttributes(ResolveInfo $info): array
+    {
+        $fields = $info->getFieldSelection(PHP_INT_MAX);
+        if (isset($fields['edges']['node'])) {
+            $fields = $fields['edges']['node'];
+        }
+
+        return $fields;
+    }
+
+    /**
+     * @throws ResourceClassNotSupportedException
+     *
+     * @return object|null
+     */
+    private function getSubresource(string $rootClass, array $rootResolvedFields, array $rootIdentifiers, string $rootProperty, string $subresourceClass, bool $isCollection)
```

Comment:
this code is also done in the `SubresourceOperationFactory`, maybe we should use the service instead to prevent side effects in the future?

Reply:
Can you point me to the duplicated snippet? And I didn't manage to find it.

Reply:
If I get that correctly you want to find the subresources identifiers etc (ie: everything you need for the `subresourceDataprovider->getSubresource` context). 

I'd recommend the use of this: https://github.com/api-platform/core/blob/a5dc79534f19db6359d6fa9b14c1e9d07434392c/src/Operation/Factory/SubresourceOperationFactory.php#L47

It's cached, and we use it in `ApiLoader` and `DocumentationNormalizer` already. In ApiLoader we build a per-operation context: https://github.com/api-platform/core/blob/091865fecf69e9c60bd77b8c6c8eb1c31bce4249/src/Bridge/Symfony/Routing/ApiLoader.php#L112